### DISCUSSION TITLE
Why mutate and mutateAsync both required?

### DISCUSSION THREAD

**Discussion by ellemedit** [TanStack/query#1478] [Q&A]
I checked `useMutation` changes in migrating react-query 3.3 from 2.6.x of my project.
I don't understand why `mutate` and `mutateAsync` both needed. Both work in asynchronous and `mutateAsync` is forward compatibility of `mutate`. duplicating same function can cause confuse developers.

Could I know why?

**Comment by TkDodo** [comment]
the discussion originated here: https://github.com/tannerlinsley/react-query/issues/1111

and there is also a summary in the PR: https://github.com/tannerlinsley/react-query/pull/1130

**Reply by ellemedit** [reply]
Thank you!
